backend/src/main/kotlin/code/nebula/cipherquest/advisor/TitleQuestionAnswerAdvisor.kt

```diff
@@ -46,43 +47,43 @@ class TitleQuestionAnswerAdvisor(
         val searchRequestToUse =
             SearchRequest
                 .from(this.searchRequest)
-                .withQuery(request.userText())
-                .withFilterExpression(doGetFilterExpression(context))
+                .query(request.userText())
+                .filterExpression(doGetFilterExpression(context))
+                .build()
 
         // 2. Search for similar documents in the vector store.
         val documents = vectorStore.similaritySearch(searchRequestToUse)
 
         if (documents != null) {
-            context[RETRIEVED_DOCUMENTS] = documents
-        }
+//            TODO: Do we need it?
+//            context[RETRIEVED_DOCUMENTS] = documents
```

Comment:
Yes. We need this code because in the `after` we would retrieve the docs from the context.

BTW this exists in the QuestionAnswerAdvisor too.